app/stock/_components/StockAddModal.tsx

```diff
@@ -0,0 +1,63 @@
+import { useItems } from "@/Contexts/ItemsContext/ItemsWrapper";
+import { Stock, itemsData } from "@/Contexts/ItemsContext/data";
+import {
+  Button,
+  Card,
+  Flex,
+  Input,
+  InputProps,
+} from "@tyanpey/urban_memories_lib";
+import { useState } from "react";
+
+export const StockAddModal = ({ closeModal }: { closeModal: () => void }) => {
+  const [newStock, setNewStock] = useState<Stock>({
+    id: new Date().getTime(),
+    itemId: 0,
+    expiryDate: new Date(),
+  });
+
+  const { stocks, setStocks } = useItems();
+
+  const handleSave = () => {
+    setStocks([...stocks, newStock]);
+    setNewStock({
+      id: new Date().getTime(),
+      itemId: 0,
+      expiryDate: new Date(),
+    });
+    closeModal();
+  };
+
+  const generateSelectItems: InputProps["selectItems"] = itemsData.map(
+    (item) => ({
+      value: item.id,
+      label: item.name,
+    })
+  );
+
+  return (
+    <Card shadowed={false}>
+      <Flex direction="v" gap="20px">
+        <Flex direction="v">
+          <Input
+            mode="select"
+            selectItems={generateSelectItems}
+            onChange={(value) => {
+              setNewStock({
+                ...newStock,
+                itemId: Number(value),
```

Comment:
According to your comment, `onChange` function returns value of the input. It converts value into a string, is this correct? Because I set value on line 33 and its a number, but when I use it in onChange its a string.

Reply:
Yes, I've seen this behaviour as well, and I am also using the similar method to work around this issue! While I'm not certain (I gotta look at the code more thoroughly) but your understanding is most likely correct!

For reference this is the code.
https://github.com/tyanpey/urban_memories_lib/blob/main/src%2FComponents%2FBase%2FInput%2FInput.tsx

Feel free to create issue on UM!